src/media/js/addon/constants.js

```diff
@@ -18,7 +21,30 @@ export const NOTE_TYPES = {
                                   color: 'gray'},
 };
 
+export const STATUS_INCOMPLETE = 'incomplete';
+export const STATUS_OBSOLETE = 'obsolete';
 export const STATUS_PENDING = 'pending';
 export const STATUS_PUBLIC = 'public';
-export const STATUS_OBSOLETE = 'obsolete';
 export const STATUS_REJECTED = 'rejected';
+
+
+export const ADDON_STATUS = {'disabled': gettext('Disabled')};
+ADDON_STATUS[STATUS_INCOMPLETE] = gettext('Incomplete');
+ADDON_STATUS[STATUS_OBSOLETE] = gettext('Obsolete');
+ADDON_STATUS[STATUS_PENDING] = gettext('Pending');
+ADDON_STATUS[STATUS_PUBLIC] = gettext('Public');
+ADDON_STATUS[STATUS_REJECTED] = gettext('Rejected');
+
+export const humanizeAddonStatus = (status, isDisabled) => {
+  if (isDisabled) {
+    return ADDON_STATUS.disabled;
+  } else if (status in ADDON_STATUS) {
+    return ADDON_STATUS[status];
+  }
+}
+
+export const humanizeVersionStatus = version => {
```

Comment:
Added this for clarity, since versions and addons both have statuses, and it'd be easy to accidentally pass a version to `humanizeAddonStatus`.